### PR TITLE
Fix tenant wise image paths for custom themes

### PR DESCRIPTION
**Pull request by TanyaM**
## Purpose
Resolves https://github.com/wso2/analytics-apim/issues/1007

## Goals
Ability to reuse the same custom theme uploaded via admin portal in Analytics as well.

## Approach
Introduce new api methods in carbon dashboards and added relevant implementation in the custom theme provider.

### CODE REVIEW

File: components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java [hunk 1]
```diff
@@ -63,15 +64,32 @@ protected void unsetConfigProvider(ConfigProvider configProvider) {
...
+                this.dashboardConfigurations.getFaviconFileName();
+        LOGGER.debug("Custom theme resources path returned via '{}' class for user: '{}.'",
+                this.getClass().getName(), username);
+        return path;
+    }
+
+    @Override
+    public String getLogoPath(String username) throws DashboardException {
+        String themeConfigResourcesPath = this.dashboardConfigurations.getThemeConfigResourcesPath();
+        if (themeConfigResourcesPath == null || themeConfigResourcesPath.isEmpty()) {
```
Comment (fazlan-nazeem) [inline components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java]: StringUtils.isEmpty will do both checks. Let's use that.

File: components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java [hunk 2]
```diff
@@ -63,15 +64,32 @@ protected void unsetConfigProvider(ConfigProvider configProvider) {
     }
 
     @Override
-    public String getPath(String username) throws DashboardException {
+    public String getFaviconPath(String username) throws DashboardException {
         String themeConfigResourcesPath = this.dashboardConfigurations.getThemeConfigResourcesPath();
         if (themeConfigResourcesPath == null || themeConfigResourcesPath.isEmpty()) {
```
Comment (fazlan-nazeem) [inline components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java]: Let's change this also to StringUtils.isEmpty

File: components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java [hunk 3]
```diff
@@ -63,15 +64,32 @@ protected void unsetConfigProvider(ConfigProvider configProvider) {
...
+    @Override
+    public String getLogoPath(String username) throws DashboardException {
+        String themeConfigResourcesPath = this.dashboardConfigurations.getThemeConfigResourcesPath();
+        if (themeConfigResourcesPath == null || themeConfigResourcesPath.isEmpty()) {
+            String error = "The themeConfigResourcesPath property cannot be found from the deployment.yaml file.";
+            LOGGER.error(error);
+            throw new DashboardException(error);
+        }
+        String tenantDomain = extractTenantDomainFromUserName(username);
+        String path = themeConfigResourcesPath + "/" + tenantDomain + IMAGES_DIR +
```
Comment (fazlan-nazeem) [inline components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java]: Maybe we should File.pathSeparatorhere to support this in windows?

File: components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java [hunk 4]
```diff
@@ -63,15 +64,32 @@ protected void unsetConfigProvider(ConfigProvider configProvider) {
...
+    public String getFaviconPath(String username) throws DashboardException {
         String themeConfigResourcesPath = this.dashboardConfigurations.getThemeConfigResourcesPath();
         if (themeConfigResourcesPath == null || themeConfigResourcesPath.isEmpty()) {
             String error = "The themeConfigResourcesPath property cannot be found from the deployment.yaml file.";
             LOGGER.error(error);
             throw new DashboardException(error);
         }
         String tenantDomain = extractTenantDomainFromUserName(username);
-        String path = themeConfigResourcesPath + "/" + tenantDomain;
+        String path = themeConfigResourcesPath + "/" + tenantDomain + IMAGES_DIR +
```
Comment (fazlan-nazeem) [inline components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java]: File.pathSeparator ?

File: components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java [hunk 5]
```diff
@@ -41,6 +41,7 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(CustomDashboardThemeConfigProvider.class);
     private static final String AT = "@";
+    private static final String IMAGES_DIR = "/analytics/images/";
```
Comment (pahans) [inline components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java]: might need `File.pathSeparator` here.

File: components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java (hunk 1 above)
Comment (TanyaM) [reply components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java]: apache lang3 is not available in this module. Hence used the above approach. 
Anyway this particular logic will be removed in the next PR.

File: components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java (hunk 2 above)
Comment (TanyaM) [reply components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java]: apache lang3 is not available in this module. Hence used the above approach.
Anyway this particular logic will be removed in the next PR.

File: components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java (hunk 3 above)
Comment (TanyaM) [reply components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java]: These are file serving URL's

File: components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java (hunk 4 above)
Comment (TanyaM) [reply components/org.wso2.analytics.apim.dashboards/org.wso2.analytics.apim.dashboards.theme.config.provider/src/main/java/org/wso2/analytics/apim/dashboards/theme/config/provider/CustomDashboardThemeConfigProvider.java]: These are file serving URL's

**Pull request merged**